### ISSUE TITLE
Radio buttons not applying attr.name when using [name] binding

### ISSUE THREAD

**Issue by JohnsonMG** [angular/angular#39460]
# 🐞 bug report

### Is this a regression?

No

### Description

When binding to radio inputs, using `[name]="myNameVariable"` does not apply the attr.name value, meaning the tab order does not work. Using only `[attr.name]="myNameVariable"` results in only the native attribute being set and so all radio groups have the same name as far as the form control is concerned.

The two solutions to this are to use both `[name]` and `[attr.name]` together, or to just use ~~name="{{myNameVariable}}"~~ (just found out while writing this that only `name="nonVariableString"` will set both the attr.name and ng-reflect, using `{{}}` interpolation doesn't work making this extra annoying) which sets both values. This is strange behaviour that isn't immediately obvious. Ideally `[name]` should also set attr.name as well as the ng-reflect.

I will be honest I am not sure what is happening behind the scenes with this component so I'm not even sure if this is possible or desired but its confusing when standard attribute names are hijacked for angular specific implementation and it prevents the native html from working.

## 🔬 Minimal Reproduction

https://plnkr.co/edit/EUv6daSYS7SC0Ncm

## 🌍  Your Environment

**Angular Version:**
I think all versions?

**Comment by AndrewKushnir** [comment]
Hi @JohnsonMG, thanks for reporting the issue.

This looks like a duplicate of #11757, so I'm closing this issue in favor of #11757 that was created earlier. Please subscribe to #11757 for further discussion and updates.

Thank you.
